phoenix-core/src/main/java/org/apache/phoenix/schema/TableRef.java

```diff
@@ -111,8 +115,10 @@ public boolean equals(Object obj) {
         if (obj == null) return false;
         if (getClass() != obj.getClass()) return false;
         TableRef other = (TableRef)obj;
+        // a null alias on either side should mean a wildcard and should not fail the equals check
         if ((alias == null && other.alias != null) || (alias != null && !alias.equals(other.alias))) return false;
-        if (!table.getName().getString().equals(other.table.getName().getString())) return false;
+        if ((table.getName() == null && other.table.getName() != null) 
```

Comment:
I think the issue is table.getName() might be null and we want to compare using  table.getName().getString()